### PR TITLE
ci: add files to run Kubernetes external storage e2e suite

### PR DESCRIPTION
**Pull request by nixpanic**
These are files that will be consumed by #1444

### CODE REVIEW

File: scripts/k8s-storage/driver-cephfs.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  FromExistingClassName: k8s-storage-e2e-cephfs
+  # FromFile: sc-cephfs.yaml
+
+SnapshotClass:
+  # Must be set to enable snapshotting tests
+  FromName: false
+
+DriverInfo:
+  # Internal name of the driver, display name in the test case and test objects
+  Name: test.cephfs.csi.ceph.com
```
Comment (Madhu-1) [inline scripts/k8s-storage/driver-cephfs.yaml:13]: as this is for logging for better clarity just call it as with cephfs.csi.ceph.com? same for rbd also

File: scripts/k8s-storage/driver-rbd-rwo.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+ShortName: cephcsi-rbd-rwo-test
```
Comment (Madhu-1) [inline scripts/k8s-storage/driver-rbd-rwo.yaml:2]: why its RWO? can we have a generic name as we keep on  adding more testing later

File: scripts/k8s-storage/driver-rbd-rwo.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+StorageClass:
+  FromExistingClassName: k8s-storage-e2e-rbd-rwo
+  # FromFile: sc-rbd.yaml
+
+SnapshotClass:
+  # Must be set to enable snapshotting tests
+  FromName: true
```
Comment (Madhu-1) [inline scripts/k8s-storage/driver-rbd-rwo.yaml:9]: need to set to false?

File: scripts/k8s-storage/sc-cephfs.yaml.in [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: k8s-storage-e2e-cephfs
+provisioner: rook-ceph.cephfs.csi.ceph.com
```
Comment (Madhu-1) [inline scripts/k8s-storage/sc-cephfs.yaml.in:6]: Need to update the provisioner name to test with ceph-csi repo

File: scripts/k8s-storage/sc-rbd.yaml.in [hunk 5]
```diff
@@ -0,0 +1,21 @@
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: k8s-storage-e2e-rbd-rwo
+provisioner: rook-ceph.rbd.csi.ceph.com
+parameters:
+   clusterID: @@CLUSTER_ID@@
+   pool: newrbdpool
```
Comment (Madhu-1) [inline scripts/k8s-storage/sc-rbd.yaml.in:9]: this should be replicapool as we create default pool as replicapool`

File: scripts/k8s-storage/driver-rbd-rwo.yaml (hunk 2 above)
Comment (nixpanic) [reply scripts/k8s-storage/driver-rbd-rwo.yaml:2]: I want to add a rwx driver too, one that disables Filesystem-mode for PVCs (not sure yet how to do that)

File: scripts/k8s-storage/driver-cephfs.yaml (hunk 1 above)
Comment (nixpanic) [reply scripts/k8s-storage/driver-cephfs.yaml:13]: sure, I can do that

File: scripts/k8s-storage/sc-cephfs.yaml.in (hunk 4 above)
Comment (nixpanic) [reply scripts/k8s-storage/sc-cephfs.yaml.in:6]: ah, right!

File: scripts/k8s-storage/driver-rbd-rwo.yaml (hunk 3 above)
Comment (nixpanic) [reply scripts/k8s-storage/driver-rbd-rwo.yaml:9]: I don't think that makes a difference, testing snapshots is disabled atm

File: scripts/k8s-storage/sc-rbd.yaml.in (hunk 5 above)
Comment (nixpanic) [reply scripts/k8s-storage/sc-rbd.yaml.in:9]: yeah, manual tests were done with a standard Rook deployment

File: scripts/k8s-storage/create-storageclasses.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+set -e
+
+WORKDIR=$(dirname "${0}")
+
+TOOLBOX_POD=$(kubectl -n rook-ceph get pods --no-headers -l app=rook-ceph-tools -o=jsonpath='{.items[0].metadata.name}')
+FS_ID=$(kubectl -n rook-ceph exec "${TOOLBOX_POD}" ceph fsid)
+
+for sc in "${WORKDIR}"/sc-*.yaml.in
+do
+	sed "s/@@CLUSTER_ID@@/${FS_ID}/" "${sc}" |
```
Comment (Madhu-1) [inline scripts/k8s-storage/create-storageclasses.sh:25]: CLUSTER_ID can be anything we just need to create the configmap with the same ID.

**Review by Madhu-1 — APPROVED** [review]

File: scripts/k8s-storage/create-storageclasses.sh (hunk 6 above)
Comment (Madhu-1) [reply scripts/k8s-storage/create-storageclasses.sh:25]: @nixpanic just discard it.

**Review by humblec — APPROVED** [review]

**Pull request merged**